spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/RedisProperties.java

```diff
@@ -67,10 +67,20 @@
 	private boolean ssl;
 
 	/**
-	 * Connection timeout.
+	 * Connection and read timeout.
 	 */
 	private Duration timeout;
 
+	/**
+	 * Connection timeout.
+	 */
+	private Duration connectionTimeout;
+
+	/**
+	 * Read timeout.
+	 */
+	private Duration readTimeout;
```

Comment:
Rather than adding two now configuration properties, let's reuse the existing timeout for the read timeout.